### DISCUSSION TITLE
how to implement a User-Defined History Output which explicitly depends on dt, time

### DISCUSSION THREAD

**Discussion by leeyp2009** [PrincetonUniversity/athena#486] [Q&A] [answered]
hi, does anyone know how to implement a user-defined history ouput function which also depends on current timestep dt and and current time? 
I know that this history output function only has two inputs: MeshBlock *pmb, int iout. But what if this history output also need information of current *dt* and *time*? pmb seems cannot point to dt and time.. 
many thanks!!

**Comment by tomidakn** [comment, marked as answer]
You can use pmb->pmy_mesh->time and dt.

**Reply by leeyp2009** [reply]
many thanks, it works.
